src/app/app.js

```diff
@@ -146,6 +147,19 @@ angular
       })
       .when('/censusdata/create', {
         templateUrl: 'app/views/censusdata/create.html',
+        //TODO: Will think about should conroller be seperated out for each of these views.
+        controller: 'CensusDataCreateController',
```

Comment:
Another behavior is that the moment ui-router loads a new page it initializes the controller as well. So technically even though controller is same, ui-router will kind of return a new instance with a clean slate. I want to wait for that scenario where it disturbs the state of data in any way and then we can decide 😊